### ISSUE TITLE
TypeError in select with missing values

### ISSUE THREAD

**Issue by aplavin** [JuliaDatabases/SQLite.jl#244]
A consistently reproducible example:
```
using SQLite, Tables

db = SQLite.DB()
DBInterface.execute(db, "create table tmp ( x TEXT )")
DBInterface.execute(db, "insert into tmp values (?)", (nothing,))
DBInterface.execute(db, "insert into tmp values (?)", (:a,))
DBInterface.execute(db, "select x from tmp") |> rowtable
```
Last line throws `ERROR: TypeError: in typeassert, expected Union{Missing, String}, got a value of type Symbol`. A null value is necessary for this error to appear.
I don't really understand the internal steps `SQLite.jl` takes here, but looks like the type detection is somewhat off.

**Comment by quinnj** [comment]
Yeah, the problem here is you've declared the column type as `TEXT`, which we respect when querying from that table; but you inserted a `Symbol`, which we should probably catch and NOT allow you to insert because it doesn't match the type of the column (`String`).

SQLite.jl doesn't automatically treat `Symbol`s as Strings, but will instead serialize them as `Symbol`s, so that when you query, you'll get a `Symbol` back. That might not be the most intuitive thing to do, but that's how it works currently.

**Comment by aplavin** [comment]
As you know, sqlite doesn't enforce any types, e.g. integers and blobs can easily end up in a `text` column. It would be useful for `SQLite.jl` to let users read the data anyway, which is not the case now: data in a table similar to my example in the first post can not be accessed at all.
